src/goto-symex/symex_goto.cpp

```diff
@@ -210,17 +210,24 @@ void goto_symext::symex_goto(statet &state)
   // merge_gotos when we visit new_state_pc
   framet::goto_state_listt &goto_state_list =
     state.call_stack().top().goto_state_map[new_state_pc];
-  goto_state_list.emplace_back(state.source, state);
 
-  symex_transition(state, state_pc, backward);
-
-  // adjust guards
+  // On an unconditional GOTO we don't need our state any more, as it will be
+  // overwritten by merge_goto. Therefore we move it onto goto_state_list
+  // instead of copying as usual.
   if(new_guard.is_true())
   {
+    goto_state_list.emplace_back(state.source, std::move(state));
```

Comment:
Definitely comment here that that `std::move` is moving our `goto_statet` superclass, rather than the whole of `state`